lib/smb2/file.rb

```diff
@@ -34,21 +48,28 @@ def initialize(tree:, create_response:)
   #
   # @return [Smb2::Packet::CloseResponse]
   def close
-    packet = Smb2::Packet::CloseRequest.new do |request|
-      request.file_id = self.create_response.file_id
-    end
+    packet = Smb2::Packet::CloseRequest.new(
+      file_id: self.create_response.file_id
+    )
 
     response = tree.send_recv(packet)
 
     Smb2::Packet::CloseResponse.new(response)
   end
 
+  # Whether the current read/write pointer is at the end of the file
+  #
+  # @return [Boolean]
+  def eof?
+    (@last_read_response && @last_read_response.header.nt_status == STATUS_END_OF_FILE) || pos == size
+  end
+
   # @return [String]
   def inspect
-    "#<Smb2::File file-id=#{create_response.file_id.unpack("H*").first} >"
+    "#<Smb2::File:#{filename} file-id=#{create_response.file_id.unpack("H*").first} >"
   end
 
-  # Call {#read_chunk} repeatedly until we get everything
+  # Call {#read_chunk} repeatedly until we get `length` bytes or hit eof.
```

Comment:
Link to `#eof?` when you say `hit eof`, so either `{#eof? hit eof}` or `hit {#eof? eof}` since it expands eof out to "end of file".  EOF may be a new abbreviation for the reader.